File: benchmark/nodejs/geometric_series.js
const start = Date.now();
const r = 0.999;
let s = 0.0;
let term = 1.0;
for (let i = 0; i < 1000000; i++) {
    s += term;
    term *= r;
}
const end = Date.now();
console.log(`Sum: ${s}`);
console.log(`Time: ${end - start}ms`);

File: benchmark/ruby/geometric_series.rb
start = Time.now.to_f * 1000
r = 0.999
s = 0.0
term = 1.0
1000000.times do
    s += term
    term *= r
end
finish = Time.now.to_f * 1000
puts "Sum: #{s}"
puts "Time: #{(finish - start).round}ms"

